### PR TITLE
[nibeheatpump] Readme for NibeGW for Arduino with Hardware and compilation hints

### PR DESCRIPTION
**Pull request by mschlenstedt**
Documentation: This is a Readme file with all the hardware hints and compilation requirements discissed in issue #13780

### CODE REVIEW

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+# NibeGW Hardware and Compiling hints
+
+
+## RS-485 Modules
+
+For setttings up a NibeGW you need a RS485 module. Most cheap modules out there are compatible with 5V voltage and therefore compatible with Arduino based hardware. Mostly you will get one of two commonly used designs:
```
Comment (paulianttila) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:6]: ProDiNo boards have RS-485 port build in, so would it make sense to say that if you board doesn't have rs-485 port you need a rs-485 module.

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+```
+ TX  RX  PIN2      5V  GND    Arduino
+ |   |    |        |    |
+ |   |    |        |    |
+ |   |    |---|    |    |
+ DI  RO   DE  RE  VCC  GND    Max485 bases module
+```
+
+
+## Ethernet Shield W5100
```
Comment (paulianttila) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:34]: ProDiNo boards have also Ethernet build in, so separated shield is not needed

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md (hunk 1 above)
Comment (mschlenstedt) [reply bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md]: Good point - added.

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md (hunk 2 above)
Comment (mschlenstedt) [reply bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:34]: Good point - added.

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+This Ethernet shield is based on Wiznet W5100 Ethernet Chip. It is supported by the Adruino Ehternet Library. No special configuration is needed, NibeGW supports this shield out of the box.
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:36]: Typo or intentional?
```suggestion
This Ethernet shield is based on Wiznet W5100 Ethernet Chip. It is supported by the Arduino Ethernet Library. No special configuration is needed, NibeGW supports this shield out of the box.
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+ProDiNo already have Ethernet included, so there's no need for a seperate Ethernet Shield. Also the ProDiNo Ethernet is supported by NibeGW out of the box.
+
+
+## Arduino Uno
+
+Arduino Uno has only one serial port which is shared with usb. So make sure to disconnect all hardware (ethernet shield, RS485 module, etc.) while uploading the compiled sketch to the Arduino. Furthermore do not use the usb port while Arduino is communicating with the Nibe heatpump.
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:43]:
```suggestion
Arduino Uno has only one serial port which is shared with USB. So make sure to disconnect all hardware (ethernet shield, RS485 module, etc.) while uploading the compiled sketch to the Arduino. Furthermore do not use the USB port while Arduino is communicating with the Nibe heatpump.
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thank you. Can I ask you to reformat the file so that there's one sentence per line? This will make it more readable in source form and easier to diff, comment etc. @paulianttila - would you like to check latest changes after your comments?

**Comment by mschlenstedt** [comment]
I changed the file to "one sentence per line" and also fixed the typos.

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 5]
```diff
@@ -33,14 +38,18 @@ Both types of modules work fine with NibeGW. The difference between the two is,
...
 
-ProDiNo already have Ethernet included, so there's no need for a seperate Ethernet Shield. Also the ProDiNo Ethernet is supported by NibeGW out of the box.
+ProDiNo already have Ethernet included, so there's no need for a seperate Ethernet Shield.
+Also the ProDiNo Ethernet is supported by NibeGW out of the box.
 
 
 ## Arduino Uno
 
-Arduino Uno has only one serial port which is shared with usb. So make sure to disconnect all hardware (ethernet shield, RS485 module, etc.) while uploading the compiled sketch to the Arduino. Furthermore do not use the usb port while Arduino is communicating with the Nibe heatpump.
+Arduino Uno has only one serial port which is shared with USB. So make sure to disconnect all hardware (ethernet shield, RS485 module, etc.) while uploading the compiled sketch to the Arduino.
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:51]: Can you divide this line also? :)

Comment (mschlenstedt) [reply bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md]: Done.

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
+# NibeGW Hardware and Compiling hints
+
+
+## RS-485 Modules
+
+For settting up a NibeGW you need a RS485 module. 
+While ProDiNo already have RS-485 support included, you need a seperate module for Arduino Uno.
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:7]:
```suggestion
While ProDiNo already have RS-485 support included, you need a separate module for Arduino Uno.
```

File: bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+```
+
+
+## Ethernet Shield W5100
+
+This Ethernet shield is based on Wiznet W5100 Ethernet Chip.
+It is supported by the Arduino Ehternet Library.
+No special configuration is needed, NibeGW supports this shield out of the box.
+
+ProDiNo already have Ethernet included, so there's no need for a seperate Ethernet Shield.
```
Comment (jlaur) [inline bundles/org.openhab.binding.nibeheatpump/contrib/NibeGW/Arduino/README.md:45]:
```suggestion
ProDiNo already have Ethernet included, so there's no need for a separate Ethernet Shield.
```

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**